Run rustfmt on `channelmanager`

joostjager opened the pull request:
After reviewing a few rustfmt PRs and observing that personal preferences regarding formatting vary, I tried to do this one in a minimal way.

I went through the diff looking for blow up that really stood out, but there wasn't much that looked outside acceptable bounds to me. If anything, most of what rustfmt does is a positive change the way I see it. Reducing line density helps with readability, for me.

There were two match arms for UpdateAddHTLC that maybe received a bit too much love, even though I think even those can go through as is. Added `#[rustfmt::skip]` nonetheless to at least avoid what I think would be the worst pain for some.

Outstanding PRs can be rebased painlessly using a variant of the script described in https://github.com/lightningdevkit/rust-lightning/pull/3749#issue-3019056916.

Update 5/27/2025: Added skips to every method.

joostjager commented:
Pushed a new take on channelmanager formatting. Used an automated script to insert the skip directives before every function.

joostjager commented on lightning/src/ln/channelmanager.rs:
```diff
@@ -5912,6 +6112,7 @@ where
 	///
 	/// Should only really ever be called in response to a PendingHTLCsForwardable event.
 	/// Will likely generate further events.
+	#[rustfmt::skip]
```
Removed. Script was a bit too enthusiastic.

joostjager commented:
Rebased after merge of #3802. Smooth workflow now in channelmanager.rs. Just remove a skip attribute and save (format on save) to go to standard formatting.

joostjager commented:
> Concept ACK, I like this as an incremental approach, and any new methods that get added will be rustfmt'd from the start. Browsing through `channelmanager.rs`, it doesn't really seem too invasive to me.

It's indeed just the structs and enums that are changing. But those weren't easy to format-fix anyway.
 
> Curious your thoughts moving forward after this --
> Do you plan to create a follow-up removing (many of?) the skips? Or should we have a policy where going forward if a method is touched in a PR then it should be formatted?

Good question. If it were up to me, I'd just format the whole repository without any skips, and use the rebase script for outstanding PRs. But as there is no unanimous support for that, I am trying to at least make it better than what it was.

I think going forward from this PR, it's up to contributors to remove skip directives as they see fit. Probably based on functions touched. If that's done in a separate initial commit, and the same is done for conflicting prs that touch the same functions, I think a rebase shouldn't be problematic.

valentinewallace commented:
I had to add `edition = "2021"` to our `rustfmt.toml` for format-on-save to work in `channelmanager.rs`. Did you experience this @joostjager? 

Without this, `cargo fmt` also does nothing and `rustfmt channelmanager.rs` errors with:

```
error[E0670]: `async fn` is not permitted in Rust 2015
     --> /Users/valentine/rust-lightning/lightning/src/ln/channelmanager.rs:11468:6
      |
11468 |     pub async fn process_pending_events_async<
      |         ^^^^^ to use `async fn`, switch to Rust 2018 or later
      |
      = help: pass `--edition 2021` to `rustc`
      = note: for more on editions, read https://doc.rust-lang.org/edition-guide

valentinewallace reviewed (dismissed):
Ship it! 👏

joostjager commented:
As discussed in sync meet, I've updated the script to drop the skip directives for functions that only need their signature to change. Added fixup commit.

joostjager commented:
Rebased. Would be nice to move towards a conclusion on this approach.

joostjager commented:
Rebased

valentinewallace commented on lightning/src/ln/channelmanager.rs:
```diff
@@ -2737,17 +2823,21 @@ struct PersistenceNotifierGuard<'a, F: FnMut() -> NotifyOption> {
 
-impl<'a> PersistenceNotifierGuard<'a, fn() -> NotifyOption> { // We don't care what the concrete F is here, it's unused
+impl<'a> PersistenceNotifierGuard<'a, fn() -> NotifyOption> {
+	// We don't care what the concrete F is here, it's unused
```
This comment should move above the `impl` block
